### PR TITLE
{171652600} Fixing internal stmt-cache leak

### PR DESCRIPTION
**Pull request by riverszhang89**
Simple change to fix a memory leak in the internal statement caching.

### CODE REVIEW

File: db/sqlinterfaces.c [hunk 1]
```diff
@@ -4539,15 +4539,21 @@ static int execute_verify_indexes(struct sqlthdstate *thd,
         clnt->has_sqliterow = 1;
         rc = verify_indexes_column_value(stmt, clnt->schema_mems);
         if (gbl_enable_internal_sql_stmt_caching) {
-            stmt_cache_add_entry(thd->stmt_cache, clnt->sql, 0, stmt, clnt);
```
Comment (nirbhayc) [inline db/sqlinterfaces.c:4542]: I think we should just finalize/free this stmt here if `stmt_cache_add_entry` returns `-1`, as `stmt_cached_add_entry` already checks for duplicates.

Comment (nirbhayc) [reply db/sqlinterfaces.c:4542]: We discussed this; the cached stmt was also removed during find() so stmt_cache_add_entry wouldn't fail.

**Review by nirbhayc — APPROVED** [review]

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Error &#x26A0;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>3/471 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_transactional_rowlocks_generated<br>physrep_ignore_table<br>pmux_sqlite_file_generated</p>

**Comment by riverszhang89** [comment]
@nirbhayc Thanks for the review!

**Pull request merged**